modules/nf-core/segemehl/align/main.nf

```diff
@@ -0,0 +1,54 @@
+process SEGEMEHL_ALIGN {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::segemehl=0.3.4" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/segemehl:0.3.4--hc2ea5fd_5':
+        'quay.io/biocontainers/segemehl:0.3.4--hc2ea5fd_5' }"
+
+    input:
+    tuple val(meta), path(reads)
+    path(fasta)
+    path(index)
+
+    output:
+    tuple val(meta), path("${meta.id}/*"), emit: results
+    path "versions.yml"                  , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def reads = meta.single_end ? "-q ${reads}" : "-q ${reads[0]} -p ${reads[1]}"
+    """
+    mkdir -p $prefix
+
+    segemehl.x \\
+        -t $task.cpus \\
+        -d $fasta \\
+        -i $index \\
+        $reads \\
+        $args \\
+        -o ${prefix}/${prefix}.sam
```

Comment:
Looks like the tool could generate either a sam or a bam output so you should avoid hardcoding it to sam here. You could parse the args, find the appropriate format, and set the extension to that value. Something like this, https://github.com/nf-core/modules/blob/8ddc73f0a37e6be2e852fd828d3effe9734ffa62/modules/nf-core/ensemblvep/main.nf#L34-L43

Reply:
Pretty neat, thanks for sharing that. Torn between using `-b` or `--bam` in the config file, the former being an actual parameter of the tool, the latter being more human readable.

Reply:
huh.. I am not sure about this, but perhaps having something like `(args.contains("-b ") || args.contains("--bam ")) ? 'bam' : 'sam'` might work?